docs/reference/fields.txt

```diff
@@ -576,93 +578,341 @@ you have three options:
    a string that match that value.
 
 
-Using Symbols Or Strings Instead Of Classes
--------------------------------------------
+.. _custom-field-types:
 
-Mongoid permits using symbols or strings instead of classes to specify the
-type of fields, for example:
+Custom Field Types
+==================
+
+You can define custom types in Mongoid and determine how they are serialized
+and deserialized. In this example, we define a new field type ``Point``, which we
+can use in our model class as follows:
 
 .. code-block:: ruby
 
-  class Order
+  class Profile
     include Mongoid::Document
-
-    field :state, type: :integer
-    # Equivalent to:
-    field :state, type: "integer"
-    # Equivalent to:
-    field :state, type: Integer
+    field :location, type: Point
   end
 
-Only standard field types as listed below can be specified using symbols or
-strings in this manner. Mongoid recognizes the following expansions:
+Then make a Ruby class to represent the type. This class must define methods
+used for MongoDB serialization and deserialization as follows:
 
-- ``:array`` => ``Array``
-- ``:big_decimal`` => ``BigDecimal``
-- ``:binary`` => ``BSON::Binary``
-- ``:boolean`` => ``Mongoid::Boolean``
-- ``:date`` => ``Date``
-- ``:date_time`` => ``DateTime``
-- ``:float`` => ``Float``
-- ``:hash`` => ``Hash``
-- ``:integer`` => ``Integer``
-- ``:object_id`` => ``BSON::ObjectId``
-- ``:range`` => ``Range``
-- ``:regexp`` => ``Regexp``
-- ``:set`` => ``Set``
-- ``:string`` => ``String``
-- ``:stringified_symbol`` => ``StringifiedSymbol``
-- ``:symbol`` => ``Symbol``
-- ``:time`` => ``Time``
+.. code-block:: ruby
 
+  class Point
 
-.. _field-default-values:
+    attr_reader :x, :y
 
-Specifying Field Default Values
--------------------------------
+    def initialize(x, y)
+      @x, @y = x, y
+    end
 
-A field can be configured to have a default value. The default value can be
-fixed, as in the following example:
+    # Converts an object of this instance into a database friendly value.
+    # In this example, we store the values in the database as array.
+    def mongoize
+      [ x, y ]
+    end
 
-.. code-block:: ruby
+    class << self
 
-  class Order
-    include Mongoid::Document
+      # Takes any possible object and converts it to how it would be
+      # stored in the database.
+      def mongoize(object)
+        case object
+        when Point then object.mongoize
+        when Hash then Point.new(object[:x], object[:y]).mongoize
+        else object
+        end
+      end
 
-    field :state, type: String, default: 'created'
+      # Get the object as it was stored in the database, and instantiate
+      # this custom class from it.
+      def demongoize(object)
+        Point.new(object[0], object[1])
```

Comment:
Do we care that this code is not representative with how we have implemented our demongoize methods? I would check that the input is an array or else return nil...

Reply:
Yes, I filed https://jira.mongodb.org/browse/MONGOID-5460 for this.